File: LC1678.py
class Solution:
    def interpret(self, command: str) -> str:
        n = len(command)
        s = []
        i = 0

        while i < n:
            if command[i] == 'G':
                s.append('G')
                i += 1
            elif command[i] == '(':
                if command[i+1] == ')':
                    s.append('o')
                    i += 2
                else:
                    s.append('al')
                    i += 4
        
        return "".join(s)

File: LC1678.cpp
class Solution {
public:
    string interpret(string command) {
        string res = "";

        int i = 0;

        while(i < command.size()){
            if(command[i] == 'G'){
                res.push_back('G');
                i++;
            }
            else if(command[i] == '('){
                if(command[i+1] == ')'){
                    res.push_back('o');
                    i += 2;
                }
                else{
                    res.push_back('a');
                    res.push_back('l');
                    i+= 4;
                }
            }
        }

        return res;
    }
};
